### ISSUE TITLE
Screen Reader Cannot Tell If Changing User Or Workspace Settings

### ISSUE THREAD

**Issue by jhomme** [microsoft/vscode#97068]
Issue Type: <b>Bug</b>

1. Open Settings.
2. Tab around to Setting Switcher buttons.
3. Tab between User and Workspace buttons. You do not hear "selected" and you cannot tell which settings you will change. Also use screen reader keys to see if the window title indicates which settings you are about to change. The window title does not change. Hit SPACE and  you cannot tell if the buttons become selected.

Note that I was working with a workspace open when observing this.

VS Code version: Code - Insiders 1.45.0-insider (d69a79b73808559a91206d73d7717ff5f798f23c, 2020-05-06T09:26:27.927Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz (4 x 2397)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.87GB (3.96GB free)|
|Process Argv||
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (10)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-toc|Ala|1.5.6
spellright|ban|3.0.50
language-hugo-vscode|bud|1.1.0
vscode-markdownlint|Dav|0.35.1
vscode-pandoc|Dou|0.0.8
vscode-hugo-snippets|fiv|0.4.1
markdown-shortcuts|mdi|0.12.0
python|ms-|2020.4.76186
vscode-hugo|rus|0.24.0
hugofy|tty|0.3.3

</details>

**Comment by roblourens** [comment]
Thanks for filing this. Do you think these buttons should be marked as radio buttons? They work like tabs or radio buttons. 

@isidorn what do you think? I could add these pretty easily https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_radio_role

**Comment by jhomme** [comment]
Hi,

I think radio button behavior would be fine. I don’t know what is possible.

 

Jim

 

==========

Jim Homme

Skype: jim.homme

FreeChess: jhomme

Twitter: jimhomme

Facebook: jimhomme

**Comment by roblourens** [comment]
I guess there are also tablist/tab roles which would be more appropriate https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_Role

**Comment by isidorn** [comment]
@roblourens I would go with the role `button` and the `aria-pressed` attribute. More details here https://w3c.github.io/aria/#button
You can experiment how NVDA behaves with that and if it will read the desired state.

**Comment by MarcoZehe** [comment]
Nope, role="button" and aria-pressed makes one into a toggle button. A toggle button is in essence a checkbox in disguise, and this would be inappropriate. Either use radio button semantics, or use tablist/tab/tabpanel, and the selected (not focused, but active) tab gets aria-selected="true", the other(s) aria-selected="false". With toggle buttons, as @isidorn suggested, you have the problem that, theoretically, both could be pressed, and that would be inappropriate here.

**Comment by isidorn** [comment]
@marcozehe thanks a lot! Makes sense.

**Comment by roblourens** [comment]
Rolling this into https://github.com/microsoft/vscode/issues/71952 for that toolbar in general
